pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -2461,6 +2461,71 @@ func TestMonitorNodeHealthUpdateNodeAndPodStatusWithLease(t *testing.T) {
 	}
 }
 
+func TestMonitorNodeHealthDeleteNodeLease(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.NodeLease, true)()
+
+	fakeNow := metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC)
+	nodeName := "node0"
+	lease := createNodeLease(nodeName, metav1.NewMicroTime(fakeNow.Time))
+	fakeNodeHandler := &testutil.FakeNodeHandler{
+		Existing: []*v1.Node{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              nodeName,
+					CreationTimestamp: fakeNow,
+				},
+			},
+		},
+		Clientset: fake.NewSimpleClientset(lease),
+	}
+	nodeController, _ := newNodeLifecycleControllerFromClient(
+		nil,
+		fakeNodeHandler,
+		5*time.Minute,
+		testRateLimiterQPS,
+		testRateLimiterQPS,
+		testLargeClusterThreshold,
+		testUnhealthyThreshold,
+		testNodeMonitorGracePeriod,
+		testNodeStartupGracePeriod,
+		testNodeMonitorPeriod,
+		false)
+	nodeController.now = func() metav1.Time { return fakeNow }
+	nodeController.recorder = testutil.NewFakeRecorder()
+	if err := nodeController.syncNodeStore(fakeNodeHandler); err != nil {
+		t.Errorf("unexpected error syncing node store: %v", err)
+	}
+	if err := nodeController.syncLeaseStore(lease); err != nil {
+		t.Fatalf("unexpected error syncing lease: %v", err)
+	}
+	if err := nodeController.monitorNodeHealth(); err != nil {
+		t.Errorf("unexpected error running monitorNodeHealth(): %v", err)
+	}
+
+	nodeController.now = func() metav1.Time { return metav1.Time{Time: fakeNow.Add(5 * time.Second)} }
+	fakeNodeHandler.Delete(nodeName, metav1.NewDeleteOptions(0))
+	if err := nodeController.syncNodeStore(fakeNodeHandler); err != nil {
+		t.Errorf("unexpected error syncing node store: %v", err)
+	}
+	if err := nodeController.monitorNodeHealth(); err != nil {
+		t.Errorf("unexpected error running monitorNodeHealth(): %v", err)
+	}
+
+	actions := fakeNodeHandler.Actions()
+	if len(actions) != 1 {
+		t.Fatalf("unexpected number of actions: %d", len(actions))
+	}
+	if actions[0].GetVerb() != "delete" {
+		t.Fatalf("unexpected action verb: %s", actions[0].GetVerb())
+	}
+	if actions[0].GetResource().Resource != "leases" {
```

Comment:
is this the place you verify the action of delete was called for resource leases ?

Reply:
We should also verify the name of the lease being deleted and if that was indeed held by the node that got deleted